packages/electron-updater/src/providers/GitHubProvider.ts

```diff
@@ -54,11 +54,40 @@ export class GitHubProvider extends BaseGitHubProvider<GithubUpdateInfo> {
     const feed = parseXml(feedXml)
     // noinspection TypeScriptValidateJSTypes
     let latestRelease = feed.element("entry", false, `No published versions on GitHub`)
-    let tag: string | null
+    let tag: string | null = null
     try {
       if (this.updater.allowPrerelease) {
-        // noinspection TypeScriptValidateJSTypes
-        tag = hrefRegExp.exec(latestRelease.element("link").attribute("href"))![1]
+        const currentChannel = this.updater?.channel || String(semver.prerelease(this.updater.currentVersion)?.[0]) ||  null;
+        for (const element of feed.getElements("entry")) {
+          // noinspection TypeScriptValidateJSTypes
+          const hrefElement = hrefRegExp.exec(element.element("link").attribute("href"))!
+
+          // If this is null then something is wrong and skip this release
+          if (hrefElement == null) continue
+
+          //Get Channel from this release
+          const hrefChannel = semver.prerelease(hrefElement[1])?.[0] || null ;
+
+          //If no channel is set by the current version, then grab latest (including prerelease)
+          if (currentChannel == null || currentChannel == 'alpha' || currentChannel == 'beta'){
+            // Skip any "custom" channels
+            if (hrefChannel != null && hrefChannel !== 'alpha' && hrefChannel !== 'beta') continue
+            // Skip alphas if in Beta Channel
+            if (currentChannel == 'beta' && hrefChannel == 'alpha') continue
+            // Get tag
+            tag = hrefChannel
+            break
+          }
```

Comment:
Quick Q. Are we still allowing Beta channel updates when the user is in Alpha channel? I see this `// Skip alphas if in Beta Channel` and am curious if we need to enforce vice-versa as well. Also, just curious, why wouldn't we want to enable custom channels?

Let's use `===` instead of `==`
Also, let's flatten this a bit:
```
const shouldFetchVersion = !currentChannel || ['alpha', 'beta'].includes(currentChannel)
const isCustomChannel = !['alpha', 'beta'].includes(hrefChannel)
const channelMismatch = currentChannel === 'beta' && hrefChannel === 'alpha'

if (shouldFetchVersion && !isCustomChannel && !channelMismatch) {
   tag = hrefChannel
   break
}
```

Reply:
For the custom channels my rational was, if you are using custom channels then you wouldn't want someone to get on that track as they might not be able to auto update to production branch without reinstalling. 

I think that also ties into the alpha to beta detail you mentioned. As it stands you can only go up, alpha -> beta -> release.  I feel like most projects would want this as going from beta to alpha could have issues that would prevent reverting or going to production. But I DO see the the opposite and it could be beneficial. I thought about having a some options like 'Stay on channel only' , 'Allow step down ' etc etc but I didn't want to introduce options that weren't already supported by the generic provider. 

I am open to feedback and want to help make the GitHub provider better.

Reply:
Makes sense.

Please slim down the current logic by using this code block below instead of the nested If's
```
const shouldFetchVersion = !currentChannel || ['alpha', 'beta'].includes(currentChannel)
const isCustomChannel = !['alpha', 'beta'].includes(hrefChannel)
const channelMismatch = currentChannel === 'beta' && hrefChannel === 'alpha'

if (shouldFetchVersion && !isCustomChannel && !channelMismatch) {
   tag = hrefChannel
   break
}
const isNextPreRelease = hrefChannel && hrefChannel === currentChannel
if (isNextPreRelease) {
   tag = hrefChannel
   break
}
```